lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -666,9 +666,20 @@ export abstract class ClientApplication {
 
         try {
             this.emitEvent(EventType.LOGOUT_START, InteractionType.Redirect, logoutRequest);
+            
+            // create logout string and navigate user window to logout.
             const authClient = await this.createAuthCodeClient(serverTelemetryManager, logoutRequest && logoutRequest.authority);
-            // create logout string and navigate user window to logout. Auth module will clear cache.
             const logoutUri: string = authClient.getLogoutUri(validLogoutRequest);
+
+            // Clear caches
+            if (validLogoutRequest.account) {
+                // Clear given account.
+                this.browserStorage.removeAccount(AccountEntity.generateAccountCacheKey(validLogoutRequest.account));
+            } else {
+                // Clear all accounts and tokens
+                await this.browserStorage.clear();
+            }
```

Comment:
This is done in msal-common in the AuthorizationCodeClient.ts class, may be redundant to have it here also?

https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/df0a58b788daba179156cff6703d25d339a11f9a/lib/msal-common/src/client/AuthorizationCodeClient.ts#L121-L127

Reply:
That block will be removed.

Reply:
Assuming you mean from `msal-common`, make sure there are no side-effects for `msal-node`

Reply:
Correct, for sure.